syne_tune/optimizer/schedulers/searchers/kde_searcher.py

```diff
@@ -0,0 +1,291 @@
+            if isinstance(hp, sp.Integer):
+                self.vartypes.append(('o', (hp.lower, hp.upper)))
+            if isinstance(hp, sp.Float):
+                self.vartypes.append(('c', 0))
+
+        self.num_min_data_points = len(self.vartypes) if num_min_data_points is None else num_min_data_points
+        assert self.num_min_data_points >= len(self.vartypes)
+
+    def to_feature(self, config):
+        def numerize(value, domain, categorical_map):
+            if isinstance(domain, sp.Categorical):
+                res = categorical_map[value] / len(domain)
+                return res
+            elif isinstance(domain, sp.Float):
+                return [(value - domain.lower) / (domain.upper - domain.lower)]
+            elif isinstance(domain, sp.Integer):
+                a = 1 / (2 * (domain.upper - domain.lower + 1))
+                b = domain.upper
+                return [(value - a) / (b - a)]
+
+        return np.hstack([
+            numerize(value=config[k], domain=v, categorical_map=self.categorical_maps.get(k, {}))
+            for k, v in self.configspace.items()
+            if isinstance(v, sp.Domain)
+        ])
+
+    def from_feature(self, feature_vector):
+        def inv_numerize(values, domain, categorical_map):
+            if not isinstance(domain, sp.Domain):
+                # constant value
+                return domain
+            else:
+                if isinstance(domain, sp.Categorical):
+                    index = int(values * len(domain))
+                    return categorical_map[index]
+                elif isinstance(domain, sp.Float):
+                        return values * (domain.upper - domain.lower) + domain.lower
+                elif isinstance(domain, sp.Integer):
+                    a = 1 / (2 * (domain.upper - domain.lower + 1))
+                    b = domain.upper
+                    return np.ceil(values * (b - a) + a)
+
+        res = {}
+        curr_pos = 0
+        for k, domain in self.configspace.items():
+            if isinstance(domain, sp.Domain):
+                res[k] = domain.cast(
+                    inv_numerize(
+                        values=feature_vector[curr_pos],
+                        domain=domain,
+                        categorical_map=self.inv_categorical_maps.get(k, {})
+                    )
+                )
+                curr_pos += 1
+            else:
+                res[k] = domain
+        return res
+
+    def configure_scheduler(self, scheduler):
+        """
+        Check that scheduler is a FIFOScheduler
+
+        Args:
+            scheduler: TaskScheduler
+                Scheduler the searcher is used with.
+
+        """
+        from syne_tune.optimizer.schedulers.fifo import FIFOScheduler
+
+        if not isinstance(scheduler, FIFOScheduler):
+            raise AssertionError("This searcher only works with FIFOScheduler. For multi-fidelity scheduler, such as "
```

Comment:
Actually, I realize now that HyperbandScheduler is a subclass of FIFOScheduler, so you can remove the 2nd part of the error message.